core/src/subgraph/registrar.rs

```diff
@@ -606,7 +632,7 @@ fn create_subgraph_version(
         }
         SubgraphVersionSwitchingMode::Synced => {
             // Just set the new version is pending, unless there is no current versions yet
-            if current_version_id_opt.is_some() {
+            if current_version_id_opt.is_some() && current_is_synced {
```

Comment:
Can this be simplified to just the 2nd condition? I think `current_is_synced` implies `current_version_id_opt.is_some()`

Reply:
Yep, good catch.